### PR TITLE
add criteria hidden and hidden search file

### PR DESCRIPTION
**Pull request by melvicaballero**
Add criteria hidden and hidden search file.

### CODE REVIEW

File: src/main/java/com/jala/search/models/SearchFile.java [hunk 1]
```diff
@@ -48,21 +48,33 @@
...
                         String extensionCriteria = criteria.getExtension();
-                        if (!nameCriteria.isEmpty() && extensionCriteria.isEmpty()) {
-                            if (nameFile.contains(nameCriteria)) {
-                                filesResult.add(file);
-                            }
-                        } else if (!extensionCriteria.isEmpty() && nameCriteria.isEmpty()) {
-                            if (extensionFile.equals(extensionCriteria)) {
-                                filesResult.add(file);
+                        boolean hiddenCriteria = criteria.isHidden();
+                        boolean hiddenYN = true;
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/SearchFile.java]: what is hiddenYN?

File: src/main/java/com/jala/search/models/SearchFile.java [hunk 2]
```diff
@@ -48,21 +48,33 @@
...
+
+                        if (hiddenCriteria && !file.isHidden()) {
+                            hiddenYN=false;
+                        }
+
+                        if (!hiddenCriteria && file.isHidden()) {
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/SearchFile.java]: this if can be of following form:
`
if ((hiddenCriteria && !file.isHidden()) || (!hiddenCriteria && file.isHidden()) ) {
      hiddenYN=false;
  }
`

File: src/main/java/com/jala/search/models/SearchFile.java [hunk 3]
```diff
@@ -48,21 +48,33 @@
...
+                            hiddenYN=false;
+                        }
+
+                        if (!nameCriteria.isEmpty()) {
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/SearchFile.java]: you can join the if with && for example 
`
if (!nameCriteria.isEmpty() && !nameFile.contains(nameCriteria)) {
    hiddenYN = false;
}
`

**Review by AKatrin — CHANGES_REQUESTED** [review]

File: src/main/java/com/jala/search/models/SearchFile.java [hunk 4]
```diff
@@ -48,19 +48,24 @@
...
                         String extensionCriteria = criteria.getExtension();
-                        if (!nameCriteria.isEmpty() && extensionCriteria.isEmpty()) {
-                            if (nameFile.contains(nameCriteria)) {
-                                filesResult.add(file);
-                            }
-                        } else if (!extensionCriteria.isEmpty() && nameCriteria.isEmpty()) {
-                            if (extensionFile.equals(extensionCriteria)) {
-                                filesResult.add(file);
+                        boolean hiddenCriteria = criteria.isHidden();
+
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/SearchFile.java]: Delete empty line

File: src/main/java/com/jala/search/models/SearchFile.java [hunk 5]
```diff
@@ -48,19 +48,24 @@
...
+                        }
+                        if (!nameCriteria.isEmpty()) {
+                            if (!nameFile.contains(nameCriteria)) {
+                                hidden = false;
                             }
-                        } else if (!extensionCriteria.isEmpty() && !nameCriteria.isEmpty()) {
-                            if (nameFile.contains(nameCriteria) && extensionFile.equals(extensionCriteria)) {
-                                filesResult.add(file);
+                        }
+                        if (!extensionCriteria.isEmpty()) {
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/SearchFile.java]: > you can join the if with && for example

**Review by AKatrin — CHANGES_REQUESTED** [review]

**Review by AKatrin — APPROVED** [review]

**Review by Cxristian — APPROVED** [review]

File: src/main/java/com/jala/search/models/AssetCommon.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * the File class is an abstract representation of file and directory
+     * pathnames and properties file.
+     * @param file as File.
+     */
+    public void loadFile(File file) {
+        super.loadFile(file);
+        //TODO load the common properties file.
+
```
Comment (Regis18) [inline src/main/java/com/jala/search/models/AssetCommon.java]: borrar espacios extras

File: src/main/java/com/jala/search/models/SearchFile.java [hunk 7]
```diff
@@ -20,58 +20,87 @@
...
-                            if (extensionFile.equals(extensionCriteria)) {
-                                filesResult.add(file);
-                            }
-                        } else if (!extensionCriteria.isEmpty() && !nameCriteria.isEmpty()) {
-                            if (nameFile.contains(nameCriteria) && extensionFile.equals(extensionCriteria)) {
-                                filesResult.add(file);
-                            }
-                        } else {
-                            filesResult.add(file);
+                        // var aux.
```
Comment (Regis18) [inline src/main/java/com/jala/search/models/SearchFile.java:54]: Delete innecesary comment

File: src/main/java/com/jala/search/models/SearchFile.java [hunk 8]
```diff
@@ -20,58 +20,87 @@
...
+                        boolean addFileToResults = true;
+
+                        if ((criteria.getHidden() == TernaryBooleanEnum.OnlyTrue) && !((Asset) asset).isHidden()) {
+                            addFileToResults = false;
+                        }
+
+                        if (addFileToResults && criteria.getHidden() == TernaryBooleanEnum.OnlyFalse && asset.isHidden()){
```
Comment (Regis18) [inline src/main/java/com/jala/search/models/SearchFile.java]: Spacio between the condition and the key

File: src/main/java/com/jala/search/models/SearchFile.java [hunk 9]
```diff
@@ -20,58 +20,87 @@
...
+
+                        if (addFileToResults && (!nameCriteria.isEmpty()) && (!nameFile.contains(nameCriteria))) {
+                            addFileToResults = false;
+                        }
+
+                        if (addFileToResults && (!extensionCriteria.isEmpty()) && (!extensionFile.equals(extensionCriteria))) {
+                            addFileToResults = false;
+                        }
+
+                        if(addFileToResults && (criteria.getReadonly() == TernaryBooleanEnum.OnlyTrue) && !(asset).isReadOnly()){
```
Comment (Regis18) [inline src/main/java/com/jala/search/models/SearchFile.java]: Put  a spacio between condition and key

File: src/main/java/com/jala/search/models/SearchFile.java [hunk 10]
```diff
@@ -20,58 +20,87 @@
...
+                            addFileToResults = false;
+                        }
+                        if(addFileToResults && (criteria.getReadonly() == TernaryBooleanEnum.OnlyFalse) && (asset).isReadOnly()){
```
Comment (Regis18) [inline src/main/java/com/jala/search/models/SearchFile.java]: Spacio between condition and key

File: src/main/java/com/jala/search/models/TernaryBooleanEnum.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+ * 2643 Av Melchor Perez de Olguin, Colquiri Sud, Cochabamba, Bolivia.
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information of
+ * Jala Foundation, ("Confidential Information").  You shall not
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with Jala Foundation.
+ */
+
```
Comment (Regis18) [inline src/main/java/com/jala/search/models/TernaryBooleanEnum.java:12]: It can be in the package of Utils?

**Review by Regis18 — APPROVED** [review]

**Pull request merged**